lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -0,0 +1,446 @@
+    if list_distributions_by_web_acl_id and web_acl_id is None:
+        module.fail_json(msg='Error web_acl_id has not been specified.')
+
+    # get distribution id from domain name alias
+    if require_distribution_id and distribution_id is None:
+        distribution_id = service_mgr.get_distribution_id_from_domain_name(domain_name_alias)
+        if not distribution_id:
+            module.fail_json(msg='Error unable to source a distribution id from domain_name_alias')
+
+    # set appropriate ansible_facts id
+    if distribution_id:
+        result = { 'ansible_facts': { 'cloudfront': { distribution_id:{} } } }
+        facts = result['ansible_facts']['cloudfront'][distribution_id]
+    elif origin_access_identity_id:
+        result = { 'ansible_facts': { 'cloudfront': { origin_access_identity_id:{} } } }
+        facts = result['ansible_facts']['cloudfront'][origin_access_identity_id]
+    elif web_acl_id:
+        result = { 'ansible_facts': { 'cloudfront': { web_acl_id:{} } } }
+        facts = result['ansible_facts']['cloudfront'][web_acl_id]
+    else:
+        result = { 'ansible_facts': { 'cloudfront': {} } }
+        facts = result['ansible_facts']['cloudfront']
+
+    # call details based on options
+    if distribution:
+        facts['distribution'] = service_mgr.get_distribution(distribution_id)
+    if distribution_config:
+        facts['distribution_config'] = service_mgr.get_distribution_config(distribution_id)
+    if origin_access_identity:
+        facts['origin_access_identity'] = service_mgr.get_origin_access_identity(origin_access_identity_id)
+    if origin_access_identity_config:
+        facts['origin_access_identity_config'] = service_mgr.get_origin_access_identity_config(origin_access_identity_id)
+    if invalidation:
+        facts['invalidation'] = service_mgr.get_invalidation(distribution_id, invalidation_id)
+    if streaming_distribution:
+        facts['streaming_distribution'] = service_mgr.get_streaming_distribution(distribution_id)
+    if streaming_distribution_config:
+        facts['streaming_distribution_config'] = service_mgr.get_streaming_distribution_config(distribution_id)
+
+    # call list based on options
+    if all_lists or list_origin_access_identities:
+        facts['list_origin_access_identities'] = service_mgr.list_origin_access_identities()
+    if all_lists or list_distributions:
+        facts['list_distributions'] = service_mgr.list_distributions()
+    if all_lists or list_streaming_distributions:
+        facts['list_streaming_distributions'] = service_mgr.list_streaming_distributions()
+    if list_distributions_by_web_acl_id:
+        facts['list_distributions_by_web_acl'] = service_mgr.list_distributions_by_web_acl()
+    if list_invalidations:
+        facts['list_invalidations'] = service_mgr.list_invalidations(distribution_id)
+
+    result['changed'] = False
+    module.exit_json(msg="Retrieved cloudfront facts.", ansible_facts=result)
```

Comment:
Hi, I found a few bugs in my code which I have since fixed:
- default to list_distributions logic was incorrect - check on if list_distributions is specified
- list_distributions_by_web_acl wasn't passing web_acl_id
- list_invalidations - wasn't specifying DistributionId for kwargs to boto3